cmake/ThirdParty.cmake

```diff
@@ -114,49 +96,50 @@ if (OPENASSETIO_ENABLE_PYTHON)
     #-------------------------------------------------------------------
     # Target to create a Python virtual environment in the install tree.
 
-    # Download core setuptools dependencies so we don't always need an
-    # online connection to local install OpenAssetIO. Need to cache the
-    # build tools explicitly due to build isolation, see:
-    # https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#build-process
-    # This also includes runtime dependencies of packages in the
-    # resources directory, whose install will otherwise fail due to
-    # --no-index.
-    openassetio_add_pip_download_target(
-        openassetio.internal.python-venv.download-deps
-        "Downloading core Python libraries"
-        wheel==0.37.1 setuptools==49.0
-        # We must upgrade pip so that the build directory is not
-        # copied to a temporary directory, since doing so could cause
-        # Windows to fail with permissions errors when concurrently
-        # installing Python packages and building the C++ component.
-        pip==22.2.2
-    )
-
     # Target to create a Python environment in the install directory.
     # If the venv already exists then this is a no-op.
-    # Requires: openassetio.internal.python-venv.download-deps
     add_custom_target(
         openassetio.internal.python-venv.create
-        COMMAND ${CMAKE_COMMAND} -E echo -- Creating Python environment in ${CMAKE_INSTALL_PREFIX}
-        COMMAND ${Python_EXECUTABLE} -m venv ${CMAKE_INSTALL_PREFIX}
+        COMMAND ${CMAKE_COMMAND} 
+        -E echo -- Creating Python environment in ${PROJECT_BINARY_DIR}/test-venv
+        COMMAND ${Python_EXECUTABLE} -m venv ${PROJECT_BINARY_DIR}/test-venv
```

Comment:
[fix] At present, if `DOPENASSETIO_ENABLE_PYTHON_TEST_VENV` is `OFF` then it uses the environment python to create the venv.  Feels like the venv we make through cmake should always use the build dependency python?  The flag should just change the _test_ code to use that venv or not.

Otherwise, you can't use the approach of setting this to OFF at configure, and manually creating the venv once with `cmake --build build --target openassetio-python-venv` so it targets the build dev python. You have to enable ENABLE_PYTHON_TEST_VENV, then build openassetio-python-venv, then disable it so that tests don't try to create it/install deps, which is a bit of a faf.

If you already have a venv/env with a custom python, you wouldn't want to use the convenience cmake target anyway? And if you did - it'd be assumed that cmake is already pointing to the python you want as a dependency?

Reply:
This should already work - we don't override `Python_EXECUTABLE` - that comes from `find_package`. Will double-check though.

Reply:
It uses `python` as `Python_EXECUTABLE` I think... see [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/649/files#diff-aa28751992bba392bf8ddcb87798c0f4b6f24dc31b6030f4237ff1f3f0f9201fR72). I noticed this as on my Mac it means it tries to do it in python 2 🤣

Reply:
Is this something we need to address? Or is it down to the user to make sure `python` means `python3.9` when they're not using out bundled environment. That'd be my feeling.

Reply:
We need to make sure that the python venv created by cmake uses the build python, otherwise the whole 'build once' workflow is extraordinarily complicated.  It does use that to make the venv, but then it doesn't use that venv to pip install the dependencies. That was the bit to fix I think. @feltech had a plan for this.

Reply:
As discussed offline, the bug was that we were attempting to pin the core venv packages (`setuptools` et al) using the external `python` (`-m pip`) rather than the venv's `python`.  This was fixed and squashed into d95c55ba780be1a01a7747f67e4a14f22b7f19ae

Reply:
This works a charm now 👍. The make-env-one-when-online-and-reuse trick makes life so much easier. Great work @elliotcmorris and @feltech. No internets for build and test for the win 🎉 .

```
cmake -S .  ... -DOPENASSETIO_ENABLE_PYTHON_TEST_VENV=OFF
cmake --build build  --target openassetio-python-venv
```
Disconnect the internets

```
. ./build/test-venv/bin/activate
ctest --test-dir build --output-on-failure
```
⏲️ 
```
100% tests passed, 0 tests failed out of 6
```